src/pages/instances/forms/ConvertToInstanceForm.tsx

```diff
@@ -0,0 +1,179 @@
+import { ChangeEvent, FC, useCallback } from "react";
+import {
+  ActionButton,
+  Button,
+  Icon,
+  Input,
+  Modal,
+  Select,
+} from "@canonical/react-components";
+import { Form } from "react-router-dom";
+import { FormikProps } from "formik";
+import { UploadState } from "types/storage";
+import ProgressBar from "components/ProgressBar";
+import { getFileExtension, humanFileSize } from "util/helpers";
+import StoragePoolSelector from "pages/storage/StoragePoolSelector";
+import {
+  ConvertToInstanceFormValues,
+  supportedVMArchOptions,
+} from "util/convertToInstance";
+import { sanitizeInstanceName, truncateInstanceName } from "util/instances";
+
+interface Props {
+  close: () => void;
+  formik: FormikProps<ConvertToInstanceFormValues>;
+  isLoading: boolean;
+  disableSubmit: boolean;
+  project: string;
+  uploadState: UploadState | null;
+}
+
+const ConvertToInstanceForm: FC<Props> = ({
+  close,
+  formik,
+  isLoading,
+  disableSubmit,
+  project,
+  uploadState,
+}) => {
+  const handleFileChange = async (e: ChangeEvent<HTMLInputElement>) => {
+    const { onChange } = formik.getFieldProps("imageFile");
+    onChange(e);
+
+    if (e.currentTarget.files) {
+      const file = e.currentTarget.files[0];
+      const suffix = "-instance";
+      const fileExtension = getFileExtension(file.name);
+      const sanitisedFileName = sanitizeInstanceName(
+        file.name.split(fileExtension)[0],
+      );
+      const instanceName = truncateInstanceName(sanitisedFileName, suffix);
+      await formik.setFieldValue("imageFile", file);
+      await formik.setFieldValue("name", instanceName);
+
+      // validate instance name
+      await formik.validateField("name");
+      void formik.setFieldTouched("name", true, true);
+      if (!formik.errors.name) {
+        formik.setFieldError("name", undefined);
+      }
+    }
+  };
+
+  const handleCloseModal = useCallback(() => {
+    formik.resetForm();
+    close();
+  }, [formik.resetForm, close]);
+
+  return (
+    <Modal
+      close={handleCloseModal}
+      className="convert-to-instance-modal"
+      title={
+        uploadState ? "Uploading image to LXD" : "Upload virtual machine image"
```

Comment:
```suggestion
        uploadState ? "Uploading to LXD" : "Upload virtual machine"
```
As discussed, maybe better copy here without the "image" part.